mlflow/go/pkg/query/parser/parser.go

```diff
@@ -41,9 +42,11 @@ func (p *parser) advance() lexer.Token {
 	return tk
 }
 
+var ErrParserError = errors.New("parser error")
+
 func (p *parser) parseIdentifier() (Identifier, error) {
 	if p.hasTokens() && p.currentTokenKind() != lexer.Identifier {
-		return Identifier{}, fmt.Errorf("Expected identifier, got %s", p.printCurrentToken())
+		return Identifier{}, fmt.Errorf("expected identifier, got %s: %w", p.printCurrentToken(), ErrParserError)
```

Comment:
I'm not too sure about how we now wrap our own errors in a new one with some text. I think that instead we should define our own error struct with an `Error()` method, and that returns the custom text. Let me know if this is unclear!

Reply:
Please give an example to clarify.